test/test_s3.py

```diff
@@ -617,6 +617,48 @@ def test_multipart_upload_with_invalid_request(self):
 
         put_body_stream.close()
 
+    def test_on_headers_callback_failure(self):
+        request = self._get_object_request(self.get_test_object_path)
+
+        s3_client = s3_client_new(False, self.region, 5 * MB)
+        s3_request = s3_client.make_request(
+            request=request,
+            type=S3RequestType.GET_OBJECT,
+            on_headers=lambda status_code, headers: (_ for _ in ()).throw(RuntimeError("Error in on_headers")),
```

Comment:
trivial: `(_ for _ in ()).throw` <-- wut

might I suggest a local function:
```
def _explode(**kwargs):
    raise RuntimeError("Error in on_headers")

...
    on_headers=_explode,
```

Reply:
Thanks, updated.